pkg/controller/applicationintegration/sqs/controller.go

```diff
@@ -0,0 +1,244 @@
+)
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (sqs.Client, error)
+}
+
+type external struct {
+	client sqs.Client
+	kube   client.Client
+}
+
+// SetupQueue adds a controller that reconciles Queue.
+func SetupQueue(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.QueueGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Queue{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.QueueGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: sqs.NewClient}),
+			managed.WithInitializers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return nil, errors.New(errNotQueue)
+	}
+
+	p := &awsv1alpha3.Provider{}
+	if err := c.kube.Get(ctx, meta.NamespacedNameOf(cr.Spec.ProviderReference), p); err != nil {
+		return nil, errors.Wrap(err, errGetProvider)
+	}
+
+	if aws.BoolValue(p.Spec.UseServiceAccount) {
+		queueClient, err := c.newClientFn(ctx, []byte{}, p.Spec.Region, awsclients.UsePodServiceAccount)
+		return &external{client: queueClient, kube: c.kube}, errors.Wrap(err, errQueueClient)
+	}
+
+	if p.GetCredentialsSecretReference() == nil {
+		return nil, errors.New(errGetProviderSecret)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.CredentialsSecretRef.Namespace, Name: p.Spec.CredentialsSecretRef.Name}
+	if err := c.kube.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	queueClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: queueClient, kube: c.kube}, errors.Wrap(err, errQueueClient)
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Queue)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotQueue)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	resAttributes, err := e.client.GetQueueAttributesRequest(&awssqs.GetQueueAttributesInput{
+		QueueUrl:       aws.String(meta.GetExternalName(cr)),
+		AttributeNames: []awssqs.QueueAttributeName{awssqs.QueueAttributeNameAll},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(sqs.IsErrorNotFound, err), errGetQueueAttributesFailed)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = v1alpha1.QueueObservation{
+		ARN: resAttributes.Attributes[v1alpha1.AttributeQueueArn],
+	}
```

Comment:
Does the `URL` field of `QueueObservation` get set anywhere?

Reply:
I have updated this this code section, was some reason it is not being reflected here.